js/runner.go

```diff
@@ -765,6 +766,26 @@ func (u *ActiveVU) RunOnce() error {
 	ctx, cancel := context.WithCancel(u.RunContext)
 	defer cancel()
 	u.moduleVUImpl.ctx = ctx
+
+	eventIterData := event.IterData{
+		Iteration:    u.iteration,
+		VUID:         u.ID,
+		ScenarioName: u.scenarioName,
+	}
+
+	emitEvent := func(evt *event.Event) func() {
+		waitDone := u.moduleVUImpl.events.local.Emit(evt)
+		return func() {
+			waitCtx, waitCancel := context.WithTimeout(u.RunContext, 30*time.Minute)
+			defer waitCancel()
+			if werr := waitDone(waitCtx); werr != nil {
+				u.state.Logger.WithError(werr).Warn()
+			}
+		}
+	}
+
```

Comment:
nit: In this case this helper lambda can be `emitAndWaitEvent` which will remove the need to call the wait function.